polars/polars-core/src/chunked_array/logical/struct_/mod.rs

```diff
@@ -173,58 +177,46 @@ impl StructChunked {
             field,
             chunks: vec![arrow_array],
             null_count: 0,
+            total_null_count: 0,
         };
         out.set_null_count();
         out
     }
 
     fn set_null_count(&mut self) {
-        let mut null_count = 0;
+        // Count both the total number of nulls and the rows where everything is null
+        let (mut null_count, mut total_null_count) = (0, 0);
         let chunks_lens = self.fields()[0].chunks().len();
 
-        // fast path
-        // we early return if a column doesn't have nulls
-        for i in 0..chunks_lens {
```

Comment:
Can we keep this fast path.

Reply:
1. It was going chunk by chunk, and if it found one chunk of a multi-chunk series that had no nulls it would abandon early. Actually, when reading this I didn't understand going per-chunk. Are the chunks in every series in a struct always aligned in this way? Is it faster than just checking the null count of the whole series?
2. If we're going to create a total null count we can't stop early. However, if it's not expensive to do a first pass where we iterate over every series and grab their null counts for the total, then we could set a fast path flag there.
3. Even if we have to iterate over everything, this new version will do no more bit manipulation than it has to.

I think the question is whether a first pass over every field that doesn't get to stop early because it has to count all the nulls, but which will avoid bit operations later is faster in aggregate. I'm going to guess "yes" since structs are likely to be much "longer" than they are "wide" so I'll go ahead and do that.

Reply:
OK, I added a pre-check/count for total nulls and whether any null rows are possible. Should be fine and there's a test so the multi-chunk bug doesn't happen again.